src/app/3-custom-directive/custom.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, ElementRef } from '@angular/core';
+
+@Directive({
+  selector: '[capitalize]'
+})
+export class CustomDirective {
+
+  constructor(private eleRef: ElementRef) {}
+
+  ngOnInit() {
+    let elementText:string = this.eleRef.nativeElement.innerText;
+    let allWords = elementText.includes(" ")? elementText.split(" "): [];
+    let buildText = "";
+
+    if(elementText.length > 0)
+    {
+      if(allWords.length > 0)
+      {
+        allWords.forEach((word)=>{
+          buildText += this.capitalizeText(word) + " ";
+        });
+      }
+      else
+      {
+        buildText = this.capitalizeText(elementText);
+      }
+    }
```

Comment:
Esta implementación se puede simplificar así:
this.eleRef.nativeElement.innerText = this.eleRef.nativeElement.innerText.split(" ").map(w => this.capitalize(w)).join(" ");

No solo quedaría más leíble y corto el código, si no que sería más eficiente puesto que de la manera en que lo hiciste siempre estás agregando un espacio al final y eso no es técnicamente correcto.